packages/dd-trace/src/remote-configuration/integrity.js

```diff
@@ -0,0 +1,66 @@
+'use strict'
+
+const crypto = require('crypto')
+const { canonicalize } = require('json-canonicalize')
+
+function findSig (signatures, rcTargetsKeyId) {
+  for (const entry of signatures) {
+    if (entry.keyid === rcTargetsKeyId) {
+      if (!entry.sig) break
+      return entry.sig
+    }
+  }
+  throw new Error(`missing signature for key ${rcTargetsKeyId}`)
+}
+
+function localAtob (str) {
+  return Buffer.from(str, 'base64').toString('ascii')
+}
+
+function toDER (hex) {
+  return Buffer.concat([
+    Buffer.from('302a300506032b6570032100', 'hex'), // Static value containing OID for algorithm
+    Buffer.from(hex, 'hex')
+  ])
+}
+
+function checkIntegrity (rcTargetsKey, rcTargetsKeyId, clientGetConfigsResponse) {
+  const targets = clientGetConfigsResponse.targets
+  if (!targets) throw new Error('no field \'targets\' in clientGetConfigsResponse object')
+
+  const targetFiles = clientGetConfigsResponse.target_files
+  if (!targetFiles) throw new Error('no field \'target_files\' in clientGetConfigsResponse object')
```

Comment:
Similarly, it's possible that there's an updated targets file but no changed or modified files for your specific tracer client.

Reply:
not sure I get it :/

Reply:
We have a concept called "client side predicates" that allow you to control what tracers receive certain config. The backend can't know ahead of time what tracer clients might possibly connect to an agent, so it can't evaluate "client side predicates" and has to generate the TUF Targets file assuming that all clients will get the file.

When a tracer connects to the agent and asks for config, the agent will evaluate the client side predicates and, depending on their values, determine that your tracer shouldn't receive a config file. So there could be a state where the TUF Targets file is newer, but client side predicates prevent your tracer from actually having any updates. The agent is going to send you the updated TUF targets file in that situation, but the `target_files` field will be empty.

I think you need to just allow for `target_files` to be empty and have that stop the integrity check. Let me know if that explanation is not clear, it's tricky and I'm not sure I did it justice with my explanation 😁

Reply:
sgtm 🙇 thank!
> I think you need to just allow for target_files to be empty and have that stop the integrity check. Let me know if that explanation is not clear, it's tricky and I'm not sure I did it justice with my explanation 😁
Will do! should we update the spec with this?